cmd/web3/main.go

```diff
@@ -360,6 +363,19 @@ func main() {
 				}
 			},
 		},
+		{
+			Name:  "generate",
+			Usage: "Generate a contract",
+			Subcommands: []cli.Command{
+				{
+					Name:  "erc20",
+					Usage: "Generate a erc20 contract",
+					Action: func(c *cli.Context) {
+						GenerateContract(ctx, "erc20", c.Args())
```

Comment:
Would flags be more clear than an ordered list of arguments?

Reply:
I tried that, but I think this way is a bit more convenient, @treeder what you think?

Reply:
Well it at least needs to be documented. How will users specify the type of erc20 they want to create?

Reply:
@jmank88 which types you propose to have? I think the main idea to provide to user some basic options (like some basic erc20, erc721 and then let them customize them)

Reply:
I have no opinion on which types, but surely capped/pausable isn't the only type we will support? Maybe it can be the default and the only one for now, but we need to be forward thinking about how to support others in a way that isn't confusing. For example: Will there be separate subcommands? Or a flag specific to the `erc20` subcommand? If they are sharing a subcommand, then would flags for the arguments make more sense? Maybe we could even infer which type of contract they want based on the flags they include. etc.

Reply:
I agree with you, but I want to make it easy to use for now or we should focus on full support of all available types - @treeder could you share your thoughts?

Reply:
There's really not too many available, 5 total looks like https://github.com/OpenZeppelin/openzeppelin-solidity/tree/master/contracts/token/ERC20

So maybe we have a flag for each of those options?  Since you can pick and choose what you want. 

eg: 

```sh
web3 generate contract --type erc20 --name MyCoolTokens --symbol COOL --capped 1000 --pausable --mintable --burnable
```

Few things to note there:

* `generate contract` in case there's some other type of generation we'd do other than contracts, like maybe code gen against abi's
* doesn't have a decimals flag since I think everyone uses 18 now, there's even an [EIP to make that mandatory](https://github.com/ethereum/EIPs/issues/724). We could of course keep it, but should be optional and default to 18.
* I specified a capped, pausable and mintable contract with capped taking a value. Generated code should include only the imports I specify, extend those and call those contructors. 

It's a bit more complicated this way, but I think necessary since it's hard to say what people would want for their token.

Reply:
It might be worth keeping the `erc20`/`erc721` subcommands instead of the `--type` flag so that each type can have a separate flag set and help menu.

Reply:
Oh ya, that makes sense.

Reply:
Maybe add short aliases to generate `g` and contracts `c` too (like angular cli, has `ng g c mycomponent`).

Reply:
Cool thanks for the feedback, everything is more clear now.